gtsam_unstable/geometry/Similarity3.cpp

```diff
@@ -85,10 +86,104 @@ Point3 Similarity3::transformFrom(const Point3& p, //
   return s_ * q;
 }
 
+Pose3 Similarity3::transformFrom(const Pose3& T) const {
+  Rot3 R = R_.compose(T.rotation());
+  Point3 t = Point3(s_ * (R_ * T.translation() + t_));
+  return Pose3(R, t);
+}
+
 Point3 Similarity3::operator*(const Point3& p) const {
   return transformFrom(p);
 }
 
+std::vector<Point3Pair> Similarity3::SubtractCentroids(const std::vector<Point3Pair>& abPointPairs, const Point3& aCentroid, const Point3& bCentroid) {
+  std::vector<Point3Pair> d_abPointPairs;
+  Point3 aPoint, bPoint;
+  for (const Point3Pair& abPair : abPointPairs) {
+    std::tie(aPoint, bPoint) = abPair;
+    Point3 da = aPoint - aCentroid;
```

Comment:
I would use abPair.first and second here as it’ll be faster. Std::tie is good if you use the variables more than once.